components/ciphertool/src/main/java/org/wso2/ciphertool/utils/Utils.java

```diff
@@ -151,6 +153,42 @@ public static void writeToPropertyFile(Properties properties, String filePath) {
         }
     }
 
+    /**
+     * Retrieve the value for the given configuration from the following order of files.
+     * 1. deployment.toml
+     * 2. default.json
+     * 3. carbon.xml
+     *
+     * @param config        configuration value
+     * @param key           key of the configuration value in default.json
+     * @param defaultMap    map that contains values from default.json
+     * @param element       element
+     * @param xPath         xpath
+     * @return  configuration value
+     */
+    public static String getValueFromConfigs(String config, String key, Map<String, Object> defaultMap,
+                                             Element element, String xPath) {
+        String value = config;
+        try {
+            // if the value is empty in deployment.toml
+            if (StringUtils.isEmpty(value)) {
+                // read from default.json
+                value = defaultMap.get(key).toString();
+            }
+            // if the value is given as a reference

```

Comment:
Shall we write these kind of comments in a single line instead of having them at two places?
`If the value is given as a reference, read configs from default.json`

Reply:
Addressed by https://github.com/wso2/cipher-tool/pull/76